File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Scenery.java
package io.github.dot166.jlib.game;

import static io.github.dot166.jlib.game.GameView.PARAMS;

import android.content.Context;
import android.widget.FrameLayout;

public class Scenery extends FrameLayout implements GameObject {
    public float z;
    public float v;
    public int h, w;

    public Scenery(Context context) {
        super(context);
    }

    @Override
    public void step(long t_ms, long dt_ms, float t, float dt) {
        setTranslationX(getTranslationX() - PARAMS.TRANSLATION_PER_SEC * dt * v);
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/game/Scenery.kt
package io.github.dot166.jlib.game

import android.content.Context
import android.widget.FrameLayout

open class Scenery(context: Context?) : FrameLayout(context!!), GameObject {
    @JvmField
    var z: Float = 0f
    @JvmField
    var v: Float = 0f
    @JvmField
    var h: Int = 0
    @JvmField
    var w: Int = 0

    override fun step(tMs: Long, dtMs: Long, t: Float, dt: Float) {
        translationX = translationX - GameView.PARAMS.TRANSLATION_PER_SEC * dt * v
    }
}
